corehq/apps/api/resources/v0_5.py

```diff
@@ -1425,3 +1425,8 @@ def obj_get(self, bundle, **kwargs):
         if not (user and user.is_member_of(domain) and user.is_active):
             return None
         return user
+
+    def prepend_urls(self):
```

Comment:
https://django-tastypie.readthedocs.io/en/latest/api.html#prepend-urls

Reply:
I'm simply telling tastypie "if you get to this URL, dispatch the detail view and not the default list view".

Since we're relying on an oauthed user to get the user details this approach works better since we don't have to pass the user ID in the url.

Reply:
Oh, Can you share why this change was needed? The usual routing didn't work with OAuth?

Reply:
@mkangia The default routing dispatches the "list" flow (which relies on the `obj_get_list` method, which we don't have for this resource). The way I did it here tells tastypie to dispatch the "detail" flow when the "list" endpoint is hit.  

> The usual routing didn't work with OAuth?

Tastypie is configured to treat URLs in the following manner:
- "<resource_url>/" means "list"
- "<resource_url>/<resource_id>" means "detail"

Since the resource we're interested in is the user (who we already know thanks to oauth), there's no point in us specifying an ID on the URL, since we can simply get the user from the request. This is the reason I told tastypie to reinterpret the "list" endpoint to mean "detail" in this case - we will never be interested in a "list" interpretation of this particular resource.

Reply:
Cool.

Can you please add that info as a comment in the code so the reasoning is available long term?